src/amr/data/particles/particles_data.hpp

```diff
@@ -185,9 +205,9 @@ namespace amr
                 }
 
                 assert(particles.size() == 0);
-                particles.resize(n_particles);
+                particles.reserve(n_particles);
                 for (std::size_t i = 0; i < n_particles; ++i)
-                    particles[i] = soa.copy(i);
+                    particles.push_back(soa.copy(i));
```

Comment:
this might want to be instrumented at some point to check if we're constructing more objects than we really need to

Reply:
specific issue? (curious: what is related with your particle counter you showed me)

Reply:
pointless copies aren't costless,  (curious: yes)